RecoTracker/LSTCore/src/alpaka/Event.dev.cc

```diff
@@ -242,29 +247,33 @@ void Event::addPixelSegmentToEvent(std::vector<unsigned int> const& hitIndices0,
   alpaka::memcpy(queue_, hitIndices3_dev, hitIndices3, size);
   alpaka::memcpy(queue_, dPhiChange_dev, dPhiChange, size);
 
-  alpaka::memcpy(queue_, segmentsBuffers_->ptIn_buf, ptIn, size);
-  alpaka::memcpy(queue_, segmentsBuffers_->ptErr_buf, ptErr, size);
-  alpaka::memcpy(queue_, segmentsBuffers_->px_buf, px, size);
-  alpaka::memcpy(queue_, segmentsBuffers_->py_buf, py, size);
-  alpaka::memcpy(queue_, segmentsBuffers_->pz_buf, pz, size);
-  alpaka::memcpy(queue_, segmentsBuffers_->etaErr_buf, etaErr, size);
-  alpaka::memcpy(queue_, segmentsBuffers_->isQuad_buf, isQuad, size);
-  alpaka::memcpy(queue_, segmentsBuffers_->eta_buf, eta, size);
-  alpaka::memcpy(queue_, segmentsBuffers_->phi_buf, phi, size);
-  alpaka::memcpy(queue_, segmentsBuffers_->charge_buf, charge, size);
-  alpaka::memcpy(queue_, segmentsBuffers_->seedIdx_buf, seedIdx, size);
-  alpaka::memcpy(queue_, segmentsBuffers_->superbin_buf, superbin, size);
-  alpaka::memcpy(queue_, segmentsBuffers_->pixelType_buf, pixelType, size);
+  SegmentsPixel segmentsPixel = segmentsDC_->view<SegmentsPixelSoA>();
+  alpaka::memcpy(queue_, alpaka::createView(devAcc_, segmentsPixel.ptIn(), size), ptIn, size);
+  alpaka::memcpy(queue_, alpaka::createView(devAcc_, segmentsPixel.ptErr(), size), ptErr, size);
+  alpaka::memcpy(queue_, alpaka::createView(devAcc_, segmentsPixel.px(), size), px, size);
+  alpaka::memcpy(queue_, alpaka::createView(devAcc_, segmentsPixel.py(), size), py, size);
+  alpaka::memcpy(queue_, alpaka::createView(devAcc_, segmentsPixel.pz(), size), pz, size);
+  alpaka::memcpy(queue_, alpaka::createView(devAcc_, segmentsPixel.etaErr(), size), etaErr, size);
+  alpaka::memcpy(queue_, alpaka::createView(devAcc_, segmentsPixel.isQuad(), size), isQuad, size);
+  alpaka::memcpy(queue_, alpaka::createView(devAcc_, segmentsPixel.eta(), size), eta, size);
+  alpaka::memcpy(queue_, alpaka::createView(devAcc_, segmentsPixel.phi(), size), phi, size);
+  alpaka::memcpy(queue_, alpaka::createView(devAcc_, segmentsPixel.charge(), size), charge, size);
+  alpaka::memcpy(queue_, alpaka::createView(devAcc_, segmentsPixel.seedIdx(), size), seedIdx, size);
+  alpaka::memcpy(queue_, alpaka::createView(devAcc_, segmentsPixel.superbin(), size), superbin, size);
+  alpaka::memcpy(queue_, alpaka::createView(devAcc_, segmentsPixel.pixelType(), size), pixelType, size);
 
   // Create source views for size and mdSize
   auto src_view_size = alpaka::createView(cms::alpakatools::host(), &size, (Idx)1u);
   auto src_view_mdSize = alpaka::createView(cms::alpakatools::host(), &mdSize, (Idx)1u);
 
-  auto dst_view_segments = alpaka::createSubView(segmentsBuffers_->nSegments_buf, (Idx)1u, (Idx)pixelModuleIndex);
+  SegmentsOccupancy segmentsOccupancy = segmentsDC_->view<SegmentsOccupancySoA>();
+  auto nSegments_view = alpaka::createView(devAcc_, segmentsOccupancy.nSegments(), (Idx)nLowerModules_ + 1);
```

Comment:
better use the `segmentsOccupancy.metadata().size()` here explicitly (same below)